src/System.Diagnostics.Process/tests/ProcessTests.Linux.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Security;
+using Xunit;
+using Xunit.NetCore.Extensions;
+
+namespace System.Diagnostics.Tests
+{
+    public partial class ProcessTests : ProcessTestBase
+    {
+        [Fact]
+        public void ProcessStart_UseShellExecuteTrue_OpenFile_ThrowsIfNoDefaultProgramInstalledSucceedsOtherwise()
+        {
+            string fileToOpen = GetTestFilePath() + ".txt";
+            File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_UseShellExecuteTrue_OpenFile_ThrowsIfNoDefaultProgramInstalledSucceedsOtherwise)}");
+
+            string[] allowedProgramsToRun = { "xdg-open", "gnome-open", "kfmclient" };
+            foreach (var program in allowedProgramsToRun)
+            {
+                if (IsProgramInstalled(program))
+                {
+                    var startInfo = new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen };
```

Comment:
I tihnk it would be reasonable, especially as this is outer loop, to write to the console which of the three you're launching. That owuld allow you to look in the official run logs, and verify that it's actually doing something -- and also whether xdg-open exists on all the images or not.